### PR TITLE
Upgrade tested Kotlin and AGP versions

### PR DESCRIPTION
**Pull request by eskatos**
Kotlin
* from 2.0.20 to 2.0.21-RC
* adding 2.1.0-Beta1

AGP
* from 8.4.0 to 8.4.2
* from 8.5.0 to 8.5.2
* from 8.6.0-dev to 8.6.1
* introducing 8.7.0 and 8.8.0-dev

Android Studio performance test was left using AGP 8.4.0 and Studio Jellyfish because of
* https://github.com/gradle/gradle/issues/30815

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test AFT AST APT

**Comment by eskatos** [comment]
@bot-gradle test AFT AST APT

**Comment by eskatos** [comment]
@bot-gradle test AFT AST APT

**Comment by eskatos** [comment]
@bot-gradle test AFT AST APT

**Comment by eskatos** [comment]
@bot-gradle test APT

**Review by ov7a — APPROVED** [review]

File: gradle/dependency-management/kotlin-versions.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Generated - Update by running `./gradlew updateKotlinVersions`
-latests=1.6.10,1.6.21,1.7.0,1.7.22,1.8.0,1.8.22,1.9.0,1.9.25,2.0.0,2.0.20
+latests=1.6.10,1.6.21,1.7.0,1.7.22,1.8.0,1.8.22,1.9.0,1.9.25,2.0.0,2.0.21-RC,2.1.0-Beta1
```
Comment (cobexer) [inline gradle/dependency-management/kotlin-versions.properties:2]: ❓When do we remove versions from this list again?
image

**Comment by cobexer** [comment]
image
image

**Pull request merged**